Added copy icon animation, fixed createdAt writing

Serge044 opened the pull request:
Added copy icon animation, fixed createdAt writing(was: created_at, now: createdAt)

zonzujiro commented on src/app/types.ts:
```diff
@@ -16,7 +16,7 @@ export type JarStatisticRecord = {
   id: number;
```
While you here, pls also fix other cases with `snake_case`, like `owner_name` in `Jar`, etc.

Serge044 commented:
![copy_icon](https://github.com/zonzujiro/korchivnyky-dashboard/assets/64706187/6ce10c2e-4d09-4b99-b963-a73f3c13cbc9)

zonzujiro commented on src/app/JarsList/JarsList.module.css:
```diff
@@ -89,3 +90,38 @@
   padding-right: 10px;
 }
+
+.copyIconClicked {
```
Class names usually named with `kebab-case`. Let's be consistent :)

zonzujiro commented on src/app/JarsList/JarsList.tsx:
```diff
@@ -53,9 +59,12 @@ const JarItem = ({ jar, isSelected, onClick }: JarItemProps) => {
-          <span className={styles.icon} onClick={copyJarLink}>
-            🔗
-          </span>
+        <img
+            src='/images/copyIcon.png'
+            alt='copy icon'
+            className={copyClicked ? styles.copyIconClicked : styles.copyIcon}
+            onClick={handleClickCopy}
+          />
```
TBH I would prefer using the emoji. They will look consistent from icon to icon, because OS keeps them that way.

Maybe 💾? Or 🔗? It's a link!

zonzujiro commented on src/app/JarsList/JarsList.tsx:
```diff
@@ -31,9 +31,15 @@ const JarItem = ({ jar, isSelected, onClick }: JarItemProps) => {
 
-  const copyJarLink = (ev: React.MouseEvent<HTMLSpanElement>) => {
+  const [copyClicked, setCopyClicked] = useState(false);
+  const handleClickCopy = (ev: React.MouseEvent<HTMLSpanElement>) => {
     ev.stopPropagation();
-    navigator.clipboard.writeText(url);
+    const timeout = setTimeout(() => {
+      navigator.clipboard.writeText(url);
+      setCopyClicked((current) => !current);
+    }, 300);
+    setCopyClicked((current) => !current);
+    return () => clearTimeout(timeout);
   };
```
No need for this syncing. You can just copy link into clipboard and start navigation after that. It won't break anything and will make code cleaner.

And in general, when I need to wait for something, I usually use this util:

```ts
await new Promise(r => setTimeout(r, 300))
```

And you can move it to separate function if you want:

```ts
const waitFor = async (ms: number) => new Promise(r => setTimeout(r, ms))
```

Smaller. Cleaner

zonzujiro commented:
First of all, thank you for PR and effort. Great job!

I have one suggestion for animation - currently it's unclear what happened when you pressed on icon. It's just became bigger and then smaller.

I would suggest to add maybe a small tooltip which will say "Посилання на банку скопійовано!" or smth like that? That will appear for small period of time and then dissapear.

zonzujiro commented:
Also, do you have an [access](https://korchivnyky-dashboard-git-fork-serge044-copya-3602e4-ititarenko.vercel.app?_vercel_share=Mza9hLLZt5D1r8pYBNYN4Mk7jUCG7F7m)? Looks like you have some problems with build

Serge044 commented:
@zonzujiro I changed the copy animation, its logic is still not perfect and I will improve it later, but for now, it is a working solution. I also changed snake_cases types to Lower camelCases, fixed typos, improved style names, and added a tooltip to copy the animation.

Please pay special attention here:

src/app/Statistics/ExpensesSection/ExpensesSection.tsx

// eslint-disable-next-line @typescript-eslint/no-unused-vars

 I did this to make a build without errors, but I understand that this is a bad practice.

Serge044 commented:
![tooltipAnimation](https://github.com/zonzujiro/korchivnyky-dashboard/assets/64706187/2d32f554-ac89-4a16-9c9d-a5ccfac66dd0)
